src/asm/fstack.cpp

```diff
@@ -107,6 +119,19 @@ FileStackNode *fstk_GetFileStack() {
 	return topNode;
 }
 
+std::shared_ptr<std::string> fstk_GetUniqueIdStr() {
+	std::shared_ptr<std::string> &str = contextStack.top().uniqueIDStr;
+
+	// If a unique ID is allowed but has not been generated yet, generate one now.
+	if (str && str->empty()) {
+		auto id = macro_NewUniqueID();
+		// TODO: bleah.
```

Comment:
This would either be addressed or at *least* properly explained what's wrong, before considering merging.

Reply:
In theory, this first `resize` call also `memset`s the buffer, despite that it's going to be overwritten right afterwards. With any luck it's going to get inlined and optimised out, but I still find the logic a little... bleah.